tests/5.0/task/test_task_in_reduction_device.c

```diff
@@ -0,0 +1,69 @@
+//===--- test_task_in_reduction_device.c ------------------------------------===//
+//
+// OpenMP API Version 5.0 Nov 2018
+//
+// This test checks the task directive with the `in_reduction` reduction
+// participation clause. It performs simple array operations which are added
+// to a reduction variable in an explcit task with the in_reduction clause.
+// This test checks the above in a target context.
+//
+////===----------------------------------------------------------------------===//
+#include <assert.h>
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ompvv.h"
+
+#define N 1024
+
+int test_task_in_reduction() {
+  OMPVV_INFOMSG("test_task_in_reduction");
+  int errors = 0;
+  int num_threads = -1;
+  int y[N];
+  int z[N];
+  int sum = 0;
+  int expected_sum = 0;
+
+  for (int i = 0; i < N; i++) {
+    y[i] = i + 1;
+    z[i] = 2*(i + 1);
+  }
+
+#pragma omp target parallel reduction(task, +: sum) num_threads(OMPVV_NUM_THREADS_DEVICE) shared(y, z, num_threads) map(tofrom: sum, y, z, num_threads)
```

Comment:
It is certainly not incorrect, but are y, z, and num_threads explicitly shared ?

Reply:
Also, y and z do not need to be copied back.

Reply:
They would be shared implicitly in this context but I figured it would be good to set them out explicitly.